utils/build_datasets.py

```diff
@@ -0,0 +1,167 @@
+    """Capture all files in provided scope and select files that match a list of formats,
+    return list of dataset items"""
+
+    all_files = []
+    walk_path = synapseutils.walk(syn, scope, ["file"])
+    for dirpath, dirname, filename in walk_path:
+        all_files = all_files + filename
+    filtered_files = [
+        file[1] for file in all_files for format in formats if file[0].endswith(format)
+    ]  # only select files of desired format
+    dataset_items = [
+        {"entityId": f, "versionNumber": syn.get(f, downloadFile=False).versionLabel}
+        for f in filtered_files
+    ]
+
+    return dataset_items
+
+
+def create_dataset_entity(syn, name: str, grant: str) -> Dataset:
+    """Create an empty Synapse Dataset using the
+    Project associated with the applicable grant number as parent.
+    Return the Dataset object."""
+
+    query = f"SELECT grantId FROM syn21918972 WHERE grantViewId='{grant}'"
+    project_id = syn.tableQuery(query).asDataFrame().iat[0, 0]
+    dataset = Dataset(name=name, parent=project_id)
+
+    return dataset
+
+
+def main():
+
+    syn = synapseclient.login()
+
+    args = get_args()
+
+    dsp, new_name = args.d, args.n
+    update_dsp_sheet = None
+    
+    if os.path.exists(dsp):
+        dsp_df = pd.read_csv(dsp, keep_default_na=False)
+        print("\nData Sharing Plan read successfully!")
+    elif "syn" in dsp:
+        dsp_df = get_table(syn, dsp)
+        print(f"Data Sharing Plan acquired from Synapse table {dsp}!")
+    else:
+        print(
+            f"❗❗❗ {dsp} is not a valid Data Sharing Plan identifier. Please check your inputs and try again."
+        )
+        exit()
+
+    if dsp_df.iat[1, 0] == "DataDSP":
+        count = 0
+        for _, row in dsp_df.iterrows():
+            grant_id = row["GrantView Key"]
+            dataset_id = row["DatasetView Key"]
+            scope_id = row["DSP Dataset Alias"]
+            dataset_name = row["DSP Dataset Name"]
+            formats = re.split(", |,", row["DSP Dataset File Formats"])
+            level = row["DSP Dataset Level"]
+            if level not in ["Metadata", "Auxiliary", "Not Applicable"]:
+                print(f"\nProcessing Dataset {dataset_name}")
+                if len(dataset_id) > 0:  # check if a Dataset entity was previously recorded 
+                    print(f"--> Accessing Dataset {dataset_id}")
+                    dataset = syn.get(dataset_id)
+                    print(f"--> {dataset_id} accessed!")
+                else:
+                    print(
+                        f"--> A new Dataset will be created for files from {scope_id}"
+                    )
+                    dataset = create_dataset_entity(syn, dataset_name, grant_id)
+                    update_dsp_sheet = True  # record the new DatasetView_id in DSP
+                    print(f"--> New Dataset created!")
+
+                if len(formats) > 0:  # only filter files if formats were specified
```

Comment:
Same feedback as above; in this case, it will return `True` so long as `formats` is not an empty list (along with the other falsy values).